### ISSUE TITLE
Inconsistent parameter naming for target architecture

### ISSUE THREAD

**Issue by nicolasbender** [buildpacks/pack#2175]
### Description
Currently, different commands use different parameter names for building for a specific architecture. Here are few examples:

[`pack buildpack new`](https://github.com/buildpacks/pack/blob/98f0eb496f3ad524f5132bd9559642edc5923fdd/internal/commands/buildpack_new_test.go#L107) uses `--targets`

[`pack builder create`](https://github.com/buildpacks/pack/blob/98f0eb496f3ad524f5132bd9559642edc5923fdd/internal/commands/builder_create_test.go#L550) uses `--target`

[`pack build`](https://github.com/buildpacks/pack/blob/98f0eb496f3ad524f5132bd9559642edc5923fdd/internal/commands/build_test.go#L157) uses `--platform`

We noticed that this leads to confusion and also issues in help and log messages like the Pro-Tip for [`pack builder create`](https://github.com/buildpacks/pack/blob/98f0eb496f3ad524f5132bd9559642edc5923fdd/internal/commands/builder_create.go#L97) which proposes `--targets` instead of the correct `--target`.

### Proposed solution
Harmonization of the parameter names leads to a consistent experience for the user. Our proposal is to use an additional `--targets` alias for all commands which is based on the naming for the multi-arch properties in the toml configuration like stated in specs for [buildpack.toml](https://github.com/buildpacks/spec/blob/main/buildpack.md), [image-extension.toml](https://github.com/buildpacks/spec/blob/main/image_extension.md) and [Stacks-Removal-RFC96](https://github.com/buildpacks/rfcs/blob/main/text/0096-remove-stacks-mixins.md), [Multi-Arch-RFC128](https://github.com/buildpacks/rfcs/blob/main/text/0128-multiarch-builders-and-package.md).

**Comment by jjbustamante** [comment]
Hi @nicolasbender , thanks for filling this issue, sorry about the confusion, we actually added `--platform` to `pack build` in the very last minute before relasing pack `0.34.x` because we noticed it behavior for M1 users could be broken. We still need to do more things for multi-platform, we will keep an eye on this.

**Comment by natalieparellano** [comment]
I agree - we need to think through carefully when we're using targets to mean the run image and when we're using targets to mean the build image. Right now they MUST be the same platform, but in a world where that isn't necessarily the case we must explain ourselves.
